modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/session.js

```diff
@@ -103,7 +103,13 @@ AUI.add(
 					instance.set('timestamp');
 
 					if (event.src === SRC) {
-						Liferay.Util.fetch(URL_BASE + 'extend_session');
+						Liferay.Util.fetch(URL_BASE + 'extend_session').then(
+							(response) => {
+								if (!response.ok) {
+									instance.expire();
```

Comment:
This is too restrictive as it prevents session to retry auto-extension if the first attempt fails. In other words, we're preventing session to get extended during the interval between `hasExpiredTimeoutOffset` becomes `true` (when we start sending requests to `extend_session`) and `hasExpired` becomes `true`. In practise, this changes the expiration logic beyond the case we want to control.

Session expiration condition has to be more restrictive, i.e. tied to the event we're producing server-side when session ids don't match. So, if we return a 500 code in the other commit (we may see which response is the appropriate if session ids don't match), we should at check it here. `!response.ok` means anything outside 200-299 response code. Ideally, we should expire only when the server-side condition is met.

@markocikos @bryceosterhaus  pls add your thoughts, this is a very delicate piece of code

Reply:
Hi @jorgediaz-lr yeah I've also took a look and none of the others looked good enough to me. I was considering using some 4xx candidates although it's debatable that here we can talk about a "client error", anyways:

- "[400 - Bad request](https://www.rfc-editor.org/rfc/rfc9110.html#name-400-bad-request)" tells server perceives a client error in the request. It's too generic so it could work
-  "[401 - Unauthorized](https://www.rfc-editor.org/rfc/rfc9110.html#name-401-unauthorized)" refers to lack of access to an actual resource targeted by the URI. In our case, the "resource" is the session itself and it's taken from the JSESSIONID cookie, not from the URI so it does not feel 100% right to me.
- "[410 - Gone](https://www.rfc-editor.org/rfc/rfc9110.html#name-410-gone)" looked good initially, however, it's the same thing as above.
- "[421 - Misdirected request](https://www.rfc-editor.org/rfc/rfc9110.html#name-421-misdirected-request)" could be used as it covers [some cases](https://www.rfc-editor.org/rfc/rfc9110.html#routing.reject) where the session mismatch could fit

That leaves us with the 5xx family (server error) for which 500 is clearly the applicable one. :man_shrugging: so unless @markocikos or @bryceosterhaus have a different perspective on this, I'm fine with 500, 400 or 421

Reply:
I have force pushed this commit https://github.com/liferay-frontend/liferay-portal/pull/2471/commits/4734b4934ef83e9a2652014950608b8c4a50a080 with the `response.status == 500` condition.